test/ec_internal_test.c

```diff
@@ -156,6 +156,54 @@ static int field_tests_ecp_mont(void)
 }
 
 #ifndef OPENSSL_NO_EC2M
+/* Test that decoding of invalid GF2m field parameters fails. */
+static int ec2m_field_sanity(void)
+{
+    int ret = 0;
+    BN_CTX *ctx = BN_CTX_new();
+    BIGNUM *p, *a, *b;
+    EC_GROUP *group1 = NULL, *group2 = NULL, *group3 = NULL;
+
+    TEST_info("Testing GF2m hardening\n");
+
+    BN_CTX_start(ctx);
+    if (!TEST_ptr(p = BN_CTX_get(ctx))
+        || !TEST_ptr(a = BN_CTX_get(ctx))
+        || !TEST_ptr(b = BN_CTX_get(ctx)))
+        goto out;
+    BN_one(a);
+    BN_one(b);
+
+    /* Even pentanomial value should be rejected */
+    BN_set_word(p, 0xf2);
+    if (!TEST_ptr_null(group1 = EC_GROUP_new_curve_GF2m(p, a, b, ctx)))
+        TEST_error("Zero constant term accepted in GF2m polynomial");
+
+    /* Odd hexanomial should also be rejected */
+    BN_set_word(p, 0xf3);
+    if (!TEST_ptr_null(group2 = EC_GROUP_new_curve_GF2m(p, a, b, ctx)))
+        TEST_error("Hexanomial accepted as GF2m polynomial");
+
+    /* Excessive polynomial degree should also be rejected */
+    BN_set_word(p, 0x71);
+    if (!BN_set_bit(p, OPENSSL_ECC_MAX_FIELD_BITS + 1))
```

Comment:
Really this should be this (sorry didn't spot this on my first round review):
```suggestion
    if (!TEST_true(BN_set_bit(p, OPENSSL_ECC_MAX_FIELD_BITS + 1)))
```
Doing it this way ensures useful test output if we ever fail that call.

Reply:
Not your fault, before the force push (in code you already reviewed) the return value of `BN_set_bit()` was not checked (assumed to succeed). So this check is new, and sure, I'm accepting the change.